### PR TITLE
Vagrant: Update dev servers to cilium/ubuntu-dev box.

### PR DESCRIPTION
**Pull request by eloycoto**
To avoid 1.2GB of disk space on download all the images for devs, a new
box was created to do not contain containers in there.

All Vagrant are now using `cilium/ubuntu-dev` and Jenkins will use
`cilium/ubuntu`

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4387)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

File: test/Vagrantfile [hunk 1]
```diff
@@ -7,8 +7,8 @@ $JOB_NAME = ENV['JOB_BASE_NAME'] || "LOCAL"
 $K8S_VERSION = ENV['K8S_VERSION'] || "1.10"
 $K8S_NODES = (ENV['K8S_NODES'] || "2").to_i
 $NFS = ENV['NFS']=="1"? true : false
-$SERVER_BOX= "cilium/ubuntu"
-$SERVER_VERSION="77"
+$SERVER_BOX = (ENV['SERVER_BOX'] || "cilium/ubuntu-dev")
```
Comment (ianvernon) [inline test/Vagrantfile]: What happens if the versions get out of sync for each box? Is this possible? Shouldn't the version be set based off of the value of `SERVER_BOX`?

Comment (eloycoto) [reply test/Vagrantfile]: Same Jenkins job, so should be always in the same version

**Comment by ianvernon** [comment]
Looks fine at face-value, but I'm holding off on reviewing further until the build passes.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
Issue with github: https://jenkins.cilium.io/job/Cilium-PR-Kubernetes-Upstream/160/console

**Comment by eloycoto** [comment]
test-upstream-k8s

**Comment by eloycoto** [comment]
Kafka Issue: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/341/testReport/junit/k8s-1/11/Non_release_k8s_versions___K8sValidatedKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies/

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
@ianvernon this I think that can be merged. could you review it?

**Comment by eloycoto** [comment]
rebased due conflict with  6477c13882a6f36684a68fc4ef2af0861054a130

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
test-upstream-k8s

**Comment by eloycoto** [comment]
timeout: 
https://jenkins.cilium.io/job/Cilium-PR-Kubernetes-Upstream/187/

**Review by ianvernon — APPROVED** [review]

**Pull request merged**